#! /usr/bin/env python

PACKAGE='auv_controls'

from dynamic_reconfigure.parameter_generator_catkin import *
from driver_base.msg import SensorLevels
from math import pi

gen = ParameterGenerator()
#Name            Type   Reconfiguration level             Description         Default Min Max
gen.add( "p_gain",          double_t, SensorLevels.RECONFIGURE_RUNNING, "P Gain",           1,    -1000,    1000 )
gen.add( "i_gain",          double_t, SensorLevels.RECONFIGURE_RUNNING, "I Gain",           0,    -1000,    1000 )
gen.add( "d_gain",          double_t, SensorLevels.RECONFIGURE_RUNNING, "D Gain",           0,    -1000,    1000 )
gen.add( "use_mod",         bool_t, SensorLevels.RECONFIGURE_RUNNING, "Modular distance instead of euclidian", False )
gen.add( "mod_val",         double_t, SensorLevels.RECONFIGURE_RUNNING, "Value for modulus",           2*pi,    0.01,    360)


exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "PID"))
